pkg/clients/network/network.go

```diff
@@ -132,8 +176,155 @@ func UpdateSubnetStatusFromAzure(v *v1alpha3.Subnet, az networkmgmt.Subnet) {
 // UpdatePublicIPAddressStatusFromAzure updates the status related to the external
 // Azure public ip address in the PublicIPAddressStatus
 func UpdatePublicIPAddressStatusFromAzure(v *v1alpha3.PublicIPAddress, az networkmgmt.PublicIPAddress) {
```

Comment:
nitpick: I usually go for a `Generate*Observation` and set the returned value to `cr.Status.AtProvider` in the controller logic so that readers don't have to come here to see what's set.